### PR TITLE
feat(preview): add styled system margin props to Preview - FE-3567

### PR DESCRIPTION
**Pull request by bulaj**
### Proposed behaviour
Added styled system margin props to Preview component.

### Current behaviour
No styled system margin props on Preview component

### Checklist

- [x] Commits follow our style guide
- [ ] <del>Screenshots are included in the PR if useful
- [ ]  <del>All themes are supported if required
- [x] Unit tests added or updated if required
- [ ]  <del>Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Testing instructions
Locally in storybook by editing stories or in codesandbox.

### CODE REVIEW

**Review by dawidzarzycki — DISMISSED** [review]

File: src/components/preview/preview.style.js [hunk 1]
```diff
@@ -18,6 +19,8 @@ const PreviewStyle = styled.span`
   & + & {
     margin-top: 3px;
   }
+
+  ${margin};
```
Comment (edleeks87) [inline src/components/preview/preview.style.js:23]: Don't think you need the `;` here

File: src/components/preview/pill.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import * as React from "react";
```
Comment (edleeks87) [inline src/components/preview/pill.d.ts:1]: This file name is wrong

**Review by edleeks87 — COMMENTED** [review]
@bulaj looks good, mainly just the `d.ts` file name that needs updating

File: src/components/preview/pill.d.ts (hunk 2 above)
Comment (bulaj) [reply src/components/preview/pill.d.ts:1]: my IDE has failed me

**Review by dawidzarzycki — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**